src/tasks/worker.ts

```
@@ -0,0 +1,263 @@
+					is_executed_with__parameter_set: task.is_executed_with__parameter_set,
+					is_scheduled_with__cron_expression:
+						task.is_scheduled_with__cron_expression,
+					priority: task.priority,
+				},
+			});
+		}
+	}
+
+	// Calculate next attempt time using exponential backoff
+	private getNextAttemptTime(attempt: number): Date | null {
+		const delay = Math.ceil(Math.exp(Math.min(10, attempt))) * 1000;
+		return new Date(Date.now() + delay);
+	}
+
+	// Poll for tasks to execute
+	private poll(): void {
+		let executed = false;
+		const handlerNames = Object.keys(this.handlers);
+		const binds = handlerNames.map((_, index) => `$${index + 1}`).join(', ');
+		sbvrUtils.db
+			.transaction(async (tx) => {
+				if (!this.canExecute()) {
+					return;
+				}
+
+				const result = await tx.executeSql(
+					`SELECT ${selectColumns}
+					FROM task AS t
+					WHERE
+						t."is executed by-handler" IN (${binds}) AND
+						t."status" = 'queued' AND
+						t."attempt count" <= t."attempt limit" AND
+						(
+							t."is scheduled to execute on-time" IS NULL OR
+							t."is scheduled to execute on-time" <= CURRENT_TIMESTAMP + INTERVAL '${Math.ceil(this.interval / 1000)} second'
+						)
+					ORDER BY
+						t."is scheduled to execute on-time" ASC,
+						t."priority" DESC,
+						t."id" ASC
+					LIMIT ${Math.max(this.concurrency - this.executing, 0)}
+					FOR UPDATE SKIP LOCKED`,
+					handlerNames,
+				);
+				if (result.rows.length === 0) {
+					return;
+				}
+
+				// Tasks found, execute them in parallel
+				await Promise.all(
+					result.rows.map(async (row) => {
+						await this.execute(tx, row as PartialTask);
+					}),
+				);
+				executed = true;
+			})
+			.catch((err) => {
+				console.error('Failed polling for tasks:', err);
+			})
+			.finally(() => {
+				setTimeout(() => this.poll(), executed ? 0 : this.interval);
+			});
+	}
+
+	// Start listening and polling for tasks
+	public start(): void {
+		sbvrUtils.db.on?.(
```

Why is the `?.` needed? I feel like if it's ever undefined then the entire task system would break and that's not something we'd want to silently ignore